Helper/WebHookHandlers/Subscription.php

```diff
@@ -114,4 +125,23 @@ private function getOrder($incrementId)
 
         return false;
     }
+
+    /**
+     * @param $incrementId
+     * @throws Exception
+     */
+    private function cancel($incrementId)
+    {
+        $orderFactory = $this->orderFactory->create();
+        $order = $orderFactory->loadByIncrementId($incrementId);
+
+        $order->cancel();
+
+        if (!$order->canCancel()) {
```

Comment:
Acho que aqui é ao contrário não?

Se `$order->canCancel()` então `$order->cancel()`
Se não ` $order->setState...`.

O que acha? :thinking:

Reply:
O método cancel() faz a verificação internamente. Se não for possível cancelar eu fecho o pedido.

Reply:
Minha preocupação nesse caso é se é retornada alguma exceção nesse `cancel` caso ele não possa ser executado?

Reply:
Eu faço dessa forma porque um pedido no Magento só poderá ser cancelado se não existir uma fatura associada a ele. 
Se eu apenas usasse o método de cancelamento o status não mudaria se já existir um invoice.
Assim se ele não puder cancelar eu fecho o pedido que é o comportamento esperado para quando existe o invoice.